src/main/kotlin/ecommerce/model/Cart.kt

```diff
@@ -1,11 +1,51 @@
 package ecommerce.model
 
+import com.fasterxml.jackson.annotation.JsonIgnore
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.OneToMany
+import jakarta.persistence.OneToOne
+import jakarta.persistence.Table
 import java.time.LocalDateTime
 
-data class Cart(
+@Entity
+@Table(name = "carts")
+class Cart(
+    @OneToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "member_id", nullable = true)
+    val member: Member? = null,
+    @OneToMany(mappedBy = "cart", cascade = [CascadeType.ALL], fetch = FetchType.LAZY, orphanRemoval = true)
+    @JsonIgnore
+    val cartItem: MutableList<CartItem> = mutableListOf(),
+    @Column(name = "quantity", nullable = false)
+    var quantity: Int = 0,
+    @Column(name = "updated_at", nullable = false)
+    val newItemAddedAt: LocalDateTime = LocalDateTime.now(),
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     val id: Long? = null,
-    val memberId: Long,
-    val productId: Long,
-    val quantity: Int,
-    val addedAt: LocalDateTime? = null,
-)
+) {
+    constructor() : this(
+        member = null,
+        cartItem = mutableListOf(),
+        quantity = 0,
+        newItemAddedAt = LocalDateTime.now(),
+    )
```

Comment:
This constructor doesn’t seem necessary! Take a look at the primary constructor above and think about why that might be. 🙂